activerecord/lib/active_record/relation/predicate_builder/range_handler.rb

```diff
@@ -36,6 +40,13 @@ def call(attribute, value)
       protected
 
         attr_reader :predicate_builder
+
+      private
+        def build_bind_with_value(attribute, value)
+          bind = predicate_builder.build_query_attribute(attribute.name, value)
+          value = Float::INFINITY unless bind
```

Comment:
I think converting all out-of-range values to infinity is still problematic. Try `9223372036854775808..10`.

I'm also worried that this class really shouldn't be doing any integer/integer-like operations. Just because something is infinite doesn't mean it's comparable with zero.

How about we push the other direction, and instead take *out* the concept of infinity? The question this code needs to ask is whether the value is above- or below range: Infinity is both of those (because we give it special treatment), but over-sized integer values are one or the other.

After thinking through a couple of possible APIs, the least bad one I can come up with is to add an `impossible_database_value?` method to AMo::Attribute, with documented return values of `false` (value is normal), `true` (value is impossible, no more information available), `:above`, and `:below` (value is greater/less than all possible values). Then values where `value_for_database` will raise RangeError should return one of the truthy values from that method.

@sgrif this is extending your nice simple low level interface in way I'm not particularly proud of; opinions / better ideas very welcome.

Reply:
> I think converting all out-of-range values to infinity is still problematic. Try `9223372036854775808..10`.

ah... I've addressed it in e80e677 anyway.

Originally this PR was written as not to add any APIs as much as possible (the converting to infinity, and `bind.nil?`).
But I agree that adding an API for this issue will make it easier to understand what problems are addressed.
So I'll try another direction to make more explicit in the code.